sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamCalciteTable.java

```diff
@@ -61,6 +65,15 @@ public RelDataType getRowType(RelDataTypeFactory typeFactory) {
     return CalciteUtils.toCalciteRowType(this.beamTable.getSchema(), typeFactory);
   }
 
+  @Override
+  public Statistic getStatistic() {
+    BeamRowCountStatistics beamStatistics =
+        beamTable.getRowCount(BeamEnumerableConverter.createPipelineOptions(pipelineOptions));
```

Comment:
This `pipelineOptions` is really `sqlShellOptions`. It is only populated in the JDBC path. For this path to work, you need to wrap this whole function in the proper class loader. Here is an example: https://github.com/apache/beam/blob/0d613080918675c5f511a20a6bb6bd3558875398/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java#L110

For the `SqlTransform.query` path you need to plumb through the pipeline options. When you do that it would be good to rename this field to clarify when each works.

See slide 15 and 16 for a block diagram of how these two paths work: https://docs.google.com/presentation/d/1ezeaxu7x4y6gnijkuQExl1XYQNWOYUjzsLAJcGhy7tQ/edit#slide=id.g33b013e8f5_1_23

Reply:
Thank you for the comment and sending the slides.